tests/test_check_types.py

```diff
@@ -164,6 +186,69 @@ def test_preferred_s_smoke(
     assert list(check.check(noise)) == []
 
 
+@given(PREFERRED_ITEMS_CASED_STRATEGY, PADDING_STRATEGY, st.text())
+def test_preferred_s_case_sensitive(
+    items: dict[str, str], padding: Padding, path: str
+) -> None:
+    """Return correct matches when ignore_case=False with case variations."""
+    check = Check(
+        check_type=types.PreferredFormsSimple(items=items, padding=padding),
+        path=path,
+        message="{} || {}",
+        ignore_case=False,
+    )
+
+    text = " ".join(items.keys())
+    check_type = types.PreferredFormsSimple(items=items, padding=padding)
+    results = list(check_type.check(text, check))
+
+    assert all(result.replacements in items.values() for result in results)
+    for i, result in enumerate(results):
+        key = list(items.keys())[i]
+        expected_replacement = items[key]
```

Comment:
Since you don't require the custom bisect and accumulation logic here, it'd be neater to do something like this.

```suggestion
    for result, (original, replacement) in zip(results, items.items()):
```